libraries/chain/db_market.cpp

```diff
@@ -711,27 +858,35 @@ bool database::check_call_orders(const asset_object& mia, bool enable_black_swan
           order_pays     = usd_to_buy;
 
           filled_call    = true;
-          if( filled_limit )
-             wlog( "Multiple limit match problem (issue 338) occurred at block #${block}", ("block",head_block_num()) );
+          if( filled_limit && maint_time <= HARDFORK_CORE_453_TIME )
+             wlog( "Multiple limit match problem (issue 453) occurred at block #${block}", ("block",head_block_num()) );
        }
 
        FC_ASSERT( filled_call || filled_limit );
+       FC_ASSERT( filled_call || filled_limit_in_loop );
 
        auto old_call_itr = call_itr;
-       if( filled_call ) ++call_itr;
+       if( filled_call && maint_time <= HARDFORK_CORE_343_TIME )
+          ++call_itr;
        // when for_new_limit_order is true, the call order is maker, otherwise the call order is taker
-       fill_order(*old_call_itr, call_pays, call_receives, match_price, for_new_limit_order );
+       fill_call_order(*old_call_itr, call_pays, call_receives, match_price, for_new_limit_order );
+       if( maint_time > HARDFORK_CORE_343_TIME )
+          call_itr = call_price_index.lower_bound( call_min );
 
        auto old_limit_itr = limit_itr;
        auto next_limit_itr = std::next( limit_itr );
-       if( filled_limit ) ++limit_itr;
-       // when for_new_limit_order is true, the limit order is taker, otherwise the limit order is maker
-       bool really_filled = fill_order(*old_limit_itr, order_pays, order_receives, true, match_price, !for_new_limit_order );
-       if( !filled_limit && really_filled )
+       if( maint_time <= HARDFORK_CORE_453_TIME )
        {
-          wlog( "Cull_small issue occurred at block #${block}", ("block",head_block_num()) );
-          limit_itr = next_limit_itr;
+          if( filled_limit ) ++limit_itr;
```

Comment:
Note: the code block here, including `++limit_itr` here and below, has been refactored in a later commit, we'll need to review again.